File: algo/implementation/cpp/gradingStudents.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> gradingStudents(vector<int> grades)
{
    vector<int>::iterator It;
    for (It = grades.begin(); It != grades.end(); It++)
    {
        if (*It >= 38 && *It % 5 >= 3)
        {
            int multiplier = *It / 5;
            *It = (multiplier + 1) * 5;
        }
    }
    return grades;
}

File: algo/implementation/js/gradingStudents.js
function gradingStudents(grades) {
    return grades.map((elem) => {
        if (elem < 38)
            return elem;
        if (elem % 5 >= 3) {
            const multiplier = Math.floor(elem / 5) + 1;
            return 5 * multiplier;
        }
        return elem;
    })
}